cpp/include/cudf/types.h

```diff
@@ -27,6 +27,7 @@ typedef enum {
     GDF_TIMESTAMP,	/**< Exact timestamp encoded with int64 since UNIX epoch (Default unit millisecond) */
     GDF_CATEGORY,
     GDF_STRING,
+    GDF_BOOL,
```

Comment:
I am not sure why we would want all GDF_INT8 code to be referenced as GDF_BOOL?  While the datatype might be the same, conceptually they are different.  

"the fact that 0 means false and anything else means true."  has always been a C standard

Reply:
>I am not sure why we would want all GDF_INT8 code to be referenced as GDF_BOOL? While the datatype might be the same, conceptually they are different.

That's not what @eyalroz was suggesting. Instead he was just saying to add a `typedef int8_t gdf_bool` and then use that typedef in the `wrapper` as 

```
using cudf_bool 	= detail::wrapper<gdf_bool, GDF_BOOL>;
```

>"the fact that 0 means false and anything else means true." has always been a C standard

While this is true, all of the `operator`s for `wrapper` don't behave this way. Thus, there's a few options:

1. Add specializations for the pertinent `operator::wrapper` for `gdf_bool` such that 0 means false and not zero means true (this is the most work)
2. Require that for `gdf_bool` that 0 means false and 1 means true, and all other values are invalid.
3. Use `typedef bool gdf_bool` which automatically enforces 2., but is problematic because of the machine dependence on `sizeof(bool)`. 
4. Add a `normalize` function similar in concept to `unwrap` in that for all types other than `gdf_bool`, it's just a no-op, but for `gdf_bool` types, it checks the value of the `gdf_bool`, if 0 it stays 0, if not zero, it changes it to `1`. You would then need to update all of the necessary wrappers `operator`s to then call `normalize` on the passed in arguments.

Reply:
I'll say I don't like the idea of keeping having `cudf_foo` and `gdf_foo`. The latter identifiers should all eventually become `cudf_foo`, and we'll be stuck with a conflict.

Also - @jrhemstad , in option (1.) you mean implementing arithmetic operators, comparison operators etc? I don't think that is really too much work.

I'm in favor of (1.) , but possibly also of (4.)

Reply:
> Also - @jrhemstad , in option (1.) you mean implementing arithmetic operators, comparison operators etc? I don't think that is really too much work.

Yes, it's not _a lot_ of work, just more than 2, 3, or 4.